File: backend/ruby_foundations/exercises/small_problems_101/easy_7/multiply_all_pairs.rb
def multiply_all_pairs(array1, array2)
  results = []
  array1.product(array2) do |item1, item2|
    results << item1 * item2
  end
  results.sort
end

puts multiply_all_pairs([2, 4], [4, 3, 1, 2]) == [2, 4, 4, 6, 8, 8, 12, 16]

File: frontend/javascript/exercises/list_processing/multiply_all_pairs.js
function multiplyAllPairs(array1, array2) {
  var result = [];
  array1.forEach(function (number1) {
    array2.forEach(function (number2) {
      result.push(number1 * number2);
    });
  });
  return result.sort(function (num1, num2) {
    return num1 - num2;
  });
}

console.log(multiplyAllPairs([2, 4], [4, 3, 1, 2])); // [2, 4, 4, 6, 8, 8, 12, 16]